internal/grpc/proto/kong/admin/service/v1/consumer_group.proto

```diff
@@ -0,0 +1,189 @@
+syntax = "proto3";
+
+package kong.admin.service.v1;
+
+import "google/api/annotations.proto";
+import "google/api/visibility.proto";
+import "kong/admin/model/v1/cluster.proto";
+import "kong/admin/model/v1/consumer.proto";
+import "kong/admin/model/v1/consumer_group.proto";
+import "kong/admin/model/v1/pagination.proto";
+
+option go_package = "github.com/kong/koko/internal/gen/kong/admin/service/v1;v1";
+
+service ConsumerGroupService {
+  rpc GetConsumerGroup(GetConsumerGroupRequest) returns (GetConsumerGroupResponse) {
+    option (google.api.http) = {get: "/v1/consumer-groups/{id}"};
+  }
+  rpc CreateConsumerGroup(CreateConsumerGroupRequest) returns (CreateConsumerGroupResponse) {
+    option (google.api.http) = {
+      post: "/v1/consumer-groups"
+      body: "item"
+    };
+  }
+  rpc UpsertConsumerGroup(UpsertConsumerGroupRequest) returns (UpsertConsumerGroupResponse) {
+    option (google.api.http) = {
+      put: "/v1/consumer-groups/{item.id}"
+      body: "item"
+    };
+  }
+  rpc DeleteConsumerGroup(DeleteConsumerGroupRequest) returns (DeleteConsumerGroupResponse) {
+    option (google.api.http) = {delete: "/v1/consumer-groups/{id}"};
+  }
+  rpc ListConsumerGroups(ListConsumerGroupsRequest) returns (ListConsumerGroupsResponse) {
+    option (google.api.http) = {get: "/v1/consumer-groups"};
+  }
+
+  rpc ListConsumerGroupMembers(ListConsumerGroupMembersRequest) returns (ListConsumerGroupMembersResponse) {
+    option (google.api.http) = {get: "/v1/consumer-groups/{id}/members"};
+  }
+  rpc CreateConsumerGroupMember(CreateConsumerGroupMemberRequest) returns (CreateConsumerGroupMemberResponse) {
+    option (google.api.http) = {post: "/v1/consumers/{consumer_id}/groups/{consumer_group_id}/members"};
```

Comment:
what about naming this `AddConsumerToGroup`? also, a POST without body feels unnatural.
maybe both PUT `/v1/consumer-groups/{consumer_group_id}/members/{consumer_id}` and a PUT `/v1/consumers/{consumer_id}/groups/{consumer_group_id}` to the same grpc method.

and both GET `/v1/consumer-groups/{consumer_group_id}/members` => `ListConsumerGroupMembers` and GET `/v1/consumers/{consumer_id}/groups` => `ListConsumerGroupsByConsumer`

i think these names and paths come from the idea of a link index (necessary on relational DB to model a many:many relationship), but for the user it should just be that a consumer has a list of groups it belongs to, and the group has a list of consumers.  the link object shouldn't be visible.

Reply:
It's generally best to stick to the standard method naming conventions, and in this case, `CreateConsumerGroupMember` adheres to that:
https://google.aip.dev/136

POST requests are not expected to be idempotent, PUT requests are. Because the same member cannot be added again, a POST request is technically proper here given standard RESTful API design patterns.

The `ListConsumerGroupsByConsumer` renaming is totally valid though, have no problem renaming that as it is more specific.

Reply:
We haven't implemented yet GET on `/v1/consumers/{consumer_id}/groups`, so I think that `ListConsumerGroupsByConsumer`, which is a totally valid name, is not necessary at the moment.